### PR TITLE
Do not add padding to 3D axis limits when limits are manually set

### PR DESCRIPTION
**Pull request by scottshambaugh**
## PR Summary
Closes https://github.com/matplotlib/matplotlib/issues/18052

```python
import matplotlib.pyplot as plt
fig, ax = plt.subplots(1, 1, subplot_kw={'projection': '3d'})
ax.set_xlim(0, 0.8)
ax.set_ylim(0, 0.8)
ax.set(xlabel='x', ylabel='y', zlabel='z')
plt.show()
```

Before:
Figure_2

After (exact limits for x and y, z retains default padded limits):
Figure_1

~~There are a ton of test image changes, gonna be a hefty commit.~~

## PR Checklist

**Documentation and Tests**
- [x] Has pytest style unit tests (and `pytest` passes)
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [N/A] New plotting related features are documented with examples.

**Release Notes**
- [N/A] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [N/A] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [N/A] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

**Comment by oscargus** [comment]
Although I think that this is really how it should be, the question is if we can just enforce it or if there should be a deprecation path? Better ping in @timhoffm directly...

**Comment by scottshambaugh** [comment]
This is changing intended behavior in some of the tests so some tweaking to do, but should be enough here right now to get an idea.

**Comment by scottshambaugh** [comment]
Talked about this in the dev call today, consensus was to make the new behavior default without deprecation, but add an rc_param to switch back to current behavior.

**Comment by scottshambaugh** [comment]
This is ready for review! Note that there are a large number (57) of baseline image edits. In all cases, these are very minor - either a 1 pixel shift in some labels, or a subpixel shift in tick locations. I believe these are likely due to accumulated floating point errors as a result of several scale conversion factors that I had to add to match the baseline formatting.

Take for example axes3d_labelpad-failed-diff.png:
axes3d_labelpad-failed-diff

Which when thresholded to maximum contrast in photoshop shows some slight subpixel shifts for the tick marks:
axes3d_labelpad-failed-diff

**Comment by scottshambaugh** [comment]
Well, it looks like the floating point theory is likely correct, because the tests are all passing locally on my machine, and the different platforms are having different tests throw image comparison errors. What's the proper way to handle this? Loosen the image comparison tolerance? I'm not sure how to examine the test images from the different CI environments if I can't recreate the issue locally.

**Comment by oscargus** [comment]
The ideal way is to add platform-specific margins, like:

https://github.com/matplotlib/matplotlib/blob/e7fd79fc39459fc067649acb6a367777f9257297/lib/matplotlib/tests/test_lines.py#L187-L189

I think one way to make it at least slightly better is to pre-compute the constants. In this way one error source is removed and the code will execute slightly faster. Always possible to use a comment to illustrate where it comes from.

File: lib/mpl_toolkits/mplot3d/axes3d.py [hunk 1]
```diff
@@ -676,28 +682,119 @@ def get_w_lims(self):
...
+    def set_ybound(self, lower=None, upper=None, view_margin=None):
+        # docstring inherited
+        self._set_bound3d(self.get_ybound, self.set_ylim, self.yaxis_inverted,
+                          lower, upper, view_margin)
+
+    def set_zbound(self, lower=None, upper=None, view_margin=None):
+        """
+        Set the lower and upper numerical bounds of the z-axis.
+        This method will honor axis inversion regardless of parameter order.
+        It will not change the autoscaling setting (`.get_autoscaley_on()`).
```
Comment (oscargus) [inline lib/mpl_toolkits/mplot3d/axes3d.py:716]:
```suggestion
        Set the lower and upper numerical bounds of the z-axis.
        
        This method will honor axis inversion regardless of parameter order.
        It will not change the autoscaling setting (`.get_autoscaley_on()`).
```

File: lib/mpl_toolkits/mplot3d/axes3d.py [hunk 2]
```diff
@@ -676,28 +682,119 @@ def get_w_lims(self):
...
+        See `~.Axes.set_ylim` for full documentation
+        """
+        return self._set_lim3d(self.yaxis, bottom, top, emit=emit, auto=auto,
+                               view_margin=view_margin, axmin=ymin, axmax=ymax)
+
+    def set_zlim(self, bottom=None, top=None, *, emit=True, auto=False,
+                 view_margin=None, zmin=None, zmax=None):
+        """
+        Set 3D z limits.
+        See `~.Axes.set_ylim` for full documentation
```
Comment (oscargus) [inline lib/mpl_toolkits/mplot3d/axes3d.py:791]:
```suggestion
        Set 3D z limits.
        
        See `~.Axes.set_ylim` for full documentation
```

File: lib/mpl_toolkits/mplot3d/axes3d.py [hunk 3]
```diff
@@ -2902,10 +2988,10 @@ def errorbar(self, x, y, z, zerr=None, yerr=None, xerr=None, fmt='',
             lower limits. In that case a caret symbol is used to indicate
             this. *lims*-arguments may be scalars, or array-likes of the same
             length as the errors. To use limits with inverted axes,
-            `~.Axes.set_xlim` or `~.Axes.set_ylim` must be called before
-            `errorbar`. Note the tricky parameter names: setting e.g.
-            *ylolims* to True means that the y-value is a *lower* limit of the
-            True value, so, only an *upward*-pointing arrow will be drawn!
+            `~.Axes.set_xlim`, `~.Axes.set_ylim`, or `~.Axes.set_ylim` must be
```
Comment (oscargus) [inline lib/mpl_toolkits/mplot3d/axes3d.py:2991]:
```suggestion
            `~.Axes.set_xlim`, `~.Axes.set_ylim`, or `~.Axes.set_zlim` must be
```
?

File: lib/mpl_toolkits/mplot3d/axes3d.py [hunk 4]
```diff
@@ -676,28 +682,119 @@ def get_w_lims(self):
...
+        See `~.Axes.set_xlim` for full documentation
+        """
+        return self._set_lim3d(self.xaxis, left, right, emit=emit, auto=auto,
+                               view_margin=view_margin, axmin=xmin, axmax=xmax)
+
+    def set_ylim(self, bottom=None, top=None, *, emit=True, auto=False,
+                 view_margin=None, ymin=None, ymax=None):
+        """
+        Set 3D y limits.
+        See `~.Axes.set_ylim` for full documentation
```
Comment (oscargus) [inline lib/mpl_toolkits/mplot3d/axes3d.py:782]:
```suggestion
        Set 3D y limits.
        
        See `~.Axes.set_ylim` for full documentation
```

File: lib/mpl_toolkits/mplot3d/axes3d.py [hunk 5]
```diff
@@ -676,28 +682,119 @@ def get_w_lims(self):
...
+        delta = (upper - lower) * view_margin
+        lower -= delta
+        upper += delta
+        return axis._set_lim(lower, upper, emit=emit, auto=auto)
+
+    def set_xlim(self, left=None, right=None, *, emit=True, auto=False,
+                 view_margin=None, xmin=None, xmax=None):
+        """
+        Set 3D x limits.
+        See `~.Axes.set_xlim` for full documentation
```
Comment (oscargus) [inline lib/mpl_toolkits/mplot3d/axes3d.py:773]:
```suggestion
        Set 3D x limits.
        
        See `~.Axes.set_xlim` for full documentation
```

File: lib/matplotlib/axis.py [hunk 6]
```diff
@@ -829,13 +829,14 @@ def _get_autoscale_on(self):
     def _set_autoscale_on(self, b):
         """
         Set whether this Axis is autoscaled when drawing or by
-        `.Axes.autoscale_view`.
+        `.Axes.autoscale_view`. If b is None, then the value is not changed.
```
Comment (oscargus) [inline lib/matplotlib/axis.py:832]:
```suggestion
        `.Axes.autoscale_view`.
        
        If *b* is None, then the value is not changed.
```
private method, but still...

File: lib/mpl_toolkits/mplot3d/axes3d.py [hunk 7]
```diff
@@ -129,7 +129,9 @@ def __init__(
 
         self.xy_viewLim = Bbox.unit()
         self.zz_viewLim = Bbox.unit()
-        self.xy_dataLim = Bbox.unit()
+        xymargin = 0.05 * 10/11  # match mpl3.7 appearance
```
Comment (oscargus) [inline lib/mpl_toolkits/mplot3d/axes3d.py:132]: Precompute

File: lib/matplotlib/mpl-data/matplotlibrc [hunk 8]
```diff
@@ -425,8 +425,9 @@
 #axes.autolimit_mode: data  # If "data", use axes.xmargin and axes.ymargin as is.
                             # If "round_numbers", after application of margins, axis
                             # limits are further expanded to the nearest "round" number.
-#polaraxes.grid: True  # display grid on polar axes
-#axes3d.grid:    True  # display grid on 3D axes
+#polaraxes.grid:    True   # display grid on polar axes
+#axes3d.grid:       True   # display grid on 3D axes
+#axes3d.automargin: False  # automatically add margin when manually setting 3D axis limits
```
Comment (oscargus) [inline lib/matplotlib/mpl-data/matplotlibrc:430]: I do not fully get if this or "classic" should be True/False, but shouldn't they be different?

File: lib/mpl_toolkits/mplot3d/axes3d.py [hunk 9]
```diff
@@ -676,28 +682,119 @@ def get_w_lims(self):
...
+        if lower is None:
+            lower = old_lower
+        if upper is None:
+            upper = old_upper
+
+        set_lim(sorted((lower, upper), reverse=bool(axis_inverted())),
+                auto=None, view_margin=view_margin)
+
+    def set_xbound(self, lower=None, upper=None, view_margin=None):
+        # docstring inherited
```
Comment (oscargus) [inline lib/mpl_toolkits/mplot3d/axes3d.py:703]: This will not have `view_margin`, so I suggest copying from `set_zbound`. Same with `set_ybound`.

File: lib/mpl_toolkits/mplot3d/axes3d.py (hunk 5 above)
Comment (oscargus) [reply lib/mpl_toolkits/mplot3d/axes3d.py:773]: Wondering if this (and y/z) also may need its own doc-string considering `view_margin`?

File: lib/matplotlib/mpl-data/matplotlibrc (hunk 8 above)
Comment (scottshambaugh) [reply lib/matplotlib/mpl-data/matplotlibrc:430]: I'm not sure I follow?

File: lib/matplotlib/testing/decorators.py [hunk 10]
```diff
@@ -134,10 +135,14 @@ def copy_baseline(self, baseline, extension):
...
-            try:
-                os.symlink(orig_expected_path, expected_fname)
-            except OSError:  # On Windows, symlink *may* be unavailable.
+            if 'microsoft-standard' in uname().release:
+                # On WSL, symlink breaks silently
                 shutil.copyfile(orig_expected_path, expected_fname)
+            else:
+                try:
+                    os.symlink(orig_expected_path, expected_fname)
+                except OSError:  # On Windows, symlink *may* be unavailable.
```
[the remark this answers is not in the document]
Comment (scottshambaugh) [reply lib/matplotlib/testing/decorators.py:144]: Ah, this is cherry-picked from https://github.com/matplotlib/matplotlib/pull/25792 in anticipation of that merging first. I'll rebase this out.

File: lib/mpl_toolkits/mplot3d/axes3d.py (hunk 5 above)
Comment (scottshambaugh) [reply lib/mpl_toolkits/mplot3d/axes3d.py]: Good catch, fixed.

File: lib/matplotlib/axis.py (hunk 6 above)
Comment (scottshambaugh) [reply lib/matplotlib/axis.py]: Yeah this is a good catch, this gets wrapped by public methods so I reverted the change.

**Comment by scottshambaugh** [comment]
Thank you for the thorough review @oscargus!

Changing to constants broke all my image tests again, so I'll have to scrub through again...

To give some context for where these constants are coming from, the original behavior when generating the view limits for 3d plots was fairly broken. There was always 1/48 of the axis range added to either side of the axis bounds when drawing the axes, however this extra "view margin" was purely visual. The calculations for the zoom level for the plot, which grid/tick lines to show, and several other spots in the code all used the non-padded axis limits for their calculation. So, when I converted the backed to use the actual axis limits everywhere, this required putting in a lot of scale factors to keep the visual appearance (23/24 = 1-2*1/48, and 25/24 = 1+2*1/48).

**Comment by scottshambaugh** [comment]
Ok, tests look good now! I think the issue was that a few of the baseline images had few enough changes that they were just under the failure threshold, but the slight variation between environments pushed it over the threshold in CI. Updating those baseline images even though they weren't failing locally fixed the issue.

Added a what's new plot as well:
image

**Comment by oscargus** [comment]
Thanks for the explanation. But assuming that the range is r, then the original bounds are b = 25/24*r, so shouldn't r = 24/25*b instead of 23/24? (I may for sure not fully be understanding the code though...)

File: lib/matplotlib/mpl-data/matplotlibrc (hunk 8 above)
Comment (oscargus) [reply lib/matplotlib/mpl-data/matplotlibrc:430]: Maybe I am not interpreting "classic" correctly, but I was just thinking that "classic" possibly should have `automargin=True`, to preserve the "classic" style?

File: lib/matplotlib/tests/test_collections.py [hunk 11]
```diff
@@ -396,6 +396,7 @@ def test_EllipseCollection():
 @image_comparison(['polycollection_close.png'], remove_text=True, style='mpl20')
 def test_polycollection_close():
     from mpl_toolkits.mplot3d import Axes3D  # type: ignore
+    plt.rcParams['axes3d.automargin'] = True
```
Comment (oscargus) [inline lib/matplotlib/tests/test_collections.py:399]: If these images are still regenerated, I would probably drop these lines? (Somehow, the only point in keeping them is to avoid having to regenerate the images.)

File: lib/matplotlib/mpl-data/matplotlibrc (hunk 8 above)
Comment (scottshambaugh) [reply lib/matplotlib/mpl-data/matplotlibrc:430]: Is 'classic' the default? I'm struggling to find where else the default style is stored. Per the comment up above that we want to "make the new behavior default without deprecation", I agree it makes sense if this is a style the user has to apply, but not if it actually is the default.

**Comment by scottshambaugh** [comment]
I dug into the scale factor more after your comment, and changing two of them to 24/25 made all the baseline images pass without updates! Thank you!

I'll note however that the scale factor in ticker I left at 23/24, because 24/25 was causing an image failure for gridline locations on one of the tests. I'm not quite sure what to make of that, but I do think it's plausibly correct.

File: lib/matplotlib/mpl-data/matplotlibrc (hunk 8 above)
Comment (oscargus) [reply lib/matplotlib/mpl-data/matplotlibrc:430]: I *think* classic is pre-2.0 (as there is an mpl20 style). But will see if I can dig it up.

File: doc/users/next_whats_new/3d_axis_limits.rst [hunk 12]
```diff
@@ -0,0 +1,17 @@
+Setting 3D axis limits now set the limits exactly
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Previously, setting the limits of a 3D axis would always add a small margin to
+the limits. Limits are now set exactly by default. The newly introduced rcparam
+``axes3d.automargin`` can be used to revert to the old behavior.
```
Comment (QuLogic) [inline doc/users/next_whats_new/3d_axis_limits.rst:6]:
```suggestion
the limits. Limits are now set exactly by default. The newly introduced
:rc:`axes3d.automargin` can be used to revert to the old behavior.
```

**Review by QuLogic — COMMENTED** [review]
I'm not sure these comments really explain why 24/25, etc.

**Comment by scottshambaugh** [comment]
@QuLogic do you mean the comments in the code, or in this thread? I'm not quite sure how to concisely describe the scale factors in more detail, as they are rather magic-constanty and I figured them out 50/50 by understanding the code and trial-and-error.

**Comment by QuLogic** [comment]
I mean the comments in the code. Saying `0.019965277777777776 = 1/48*23/24` doesn't really seem to explain much more than saying `0.01996... matches mpl3.7 appearance`, as I have no idea where 48, 23, or 24 came from. Is the old padding 1/24 and every other number is ±1 or so?

**Comment by scottshambaugh** [comment]
Essentially - the old padding was 1/48, and was applied 2x to the min and max limits, so that's where the 1 ± 1/24 scaling came from. The old padding was from these lines: https://github.com/matplotlib/matplotlib/blob/fff2a79ce22ab39bea95067d30438bfdbe47f560/lib/mpl_toolkits/mplot3d/axis3d.py#L230_L232

```python
        deltas = (maxs - mins) / 12
        mins -= 0.25 * deltas
        maxs += 0.25 * deltas
```

I can take out the extra detail in the code if you think it's going to be confusing for people.

File: lib/matplotlib/mpl-data/matplotlibrc (hunk 8 above)
Comment (oscargus) [reply lib/matplotlib/mpl-data/matplotlibrc:430]: Now, I've thought a bit about it and I think it makes sense to set it to True in the classic style sheet. In that way, using that style sheet will have the pre-2.0 look. It only affects a few of the tests (the ones that to not use `style='mpl20'`). 

(If I understand things.)

File: lib/mpl_toolkits/mplot3d/tests/test_axes3d.py [hunk 13]
```diff
@@ -207,6 +207,7 @@ def test_bar3d_lightsource():
 
 @mpl3d_image_comparison(['contour3d.png'], style='mpl20')
 def test_contour3d():
+    plt.rcParams['axes3d.automargin'] = True
```
Comment (oscargus) [inline lib/mpl_toolkits/mplot3d/tests/test_axes3d.py:210]:
```suggestion
    # Remove when regenerating the image
    plt.rcParams['axes3d.automargin'] = True
```
and most below...

File: lib/mpl_toolkits/mplot3d/tests/test_axes3d.py [hunk 14]
```diff
@@ -262,6 +266,7 @@ def test_contourf3d_fill():
                                             ['max', [0, 2, 4, 6]]])
 @check_figures_equal(extensions=["png"])
 def test_contourf3d_extend(fig_test, fig_ref, extend, levels):
+    plt.rcParams['axes3d.automargin'] = True
```
Comment (oscargus) [inline lib/mpl_toolkits/mplot3d/tests/test_axes3d.py:269]:
```suggestion
```
Change in classic.mplstyle instead.

File: lib/mpl_toolkits/mplot3d/tests/test_axes3d.py [hunk 15]
```diff
@@ -692,6 +705,7 @@ def test_text3d():
 
 @check_figures_equal(extensions=['png'])
 def test_text3d_modification(fig_ref, fig_test):
+    plt.rcParams['axes3d.automargin'] = True
```
Comment (oscargus) [inline lib/mpl_toolkits/mplot3d/tests/test_axes3d.py:708]:
```suggestion
```

File: lib/mpl_toolkits/mplot3d/tests/test_axes3d.py [hunk 16]
```diff
@@ -1441,6 +1461,7 @@ def test_ax3d_tickcolour():
 
 @check_figures_equal(extensions=["png"])
 def test_ticklabel_format(fig_test, fig_ref):
+    plt.rcParams['axes3d.automargin'] = True
```
Comment (oscargus) [inline lib/mpl_toolkits/mplot3d/tests/test_axes3d.py:1464]:
```suggestion
```

File: lib/mpl_toolkits/mplot3d/tests/test_axes3d.py [hunk 17]
```diff
@@ -1158,6 +1177,7 @@ def test_world():
 def test_autoscale():
     fig, ax = plt.subplots(subplot_kw={"projection": "3d"})
     assert ax.get_zscale() == 'linear'
+    ax._view_margin = 0
```
Comment (oscargus) [inline lib/mpl_toolkits/mplot3d/tests/test_axes3d.py:1180]: Why is this required? Will skipping this lead to problems? It seems like this is a potential behavior change?

Comment (scottshambaugh) [reply lib/mpl_toolkits/mplot3d/tests/test_axes3d.py:1178]: Previously, the 1/48 view margin on both the upper and lower sides of an axis was applied in rendering only, so that an axis limit in the range (0, 1) would report its limits from `get_w_lims()` as (0, 1), but visually show as (-0.020833333333333332, 1.020833333333333332) = (-1/48, 49/48). This PR does change that behavior in that `get_w_lims()` reports the actual rendered limits instead of the "set" ones. I would consider the previous behavior a bug, but this might be breaking someone's workflow somewhere. Thoughts on how to handle that?

File: lib/matplotlib/mpl-data/matplotlibrc (hunk 8 above)
Comment (tacaswell) [reply lib/matplotlib/mpl-data/matplotlibrc:430]: Yes, the "classic" style sheet is from the 2.0 release which we provided so that users who wanted to go back _can_ and we use it internally (plus some extra settings) in our tests to avoid having to re-generate a lot of images.

**Comment by tacaswell** [comment]
@scottshambaugh this needs a rebase because we merged some other PRs

**Comment by tacaswell** [comment]
Talking about this with @ksunden and @QuLogic we want to take this, but given this is an API change and it is the 11th hour of the 3.8 prep, we are going to hold this for 3.9 (which means it should get merged as soon we branch).

**Comment by QuLogic** [comment]
> Essentially - the old padding was 1/48, and was applied 2x (to the min and max limits), so that's where the 1 ± 1/24 scaling came from.

OK, can we mention that in the comments? It currently says something like "to match mpl3.7 appearance", so it can say "to match mpl3.7 appearance that added 1/48 padding". Also feel free to put a longer explanation in the commit message.

**Comment by scottshambaugh** [comment]
Thanks! Rebased/squashed, and added a lot more detail to the comments on scale factors. @QuLogic do those address your concerns?

**Comment by scottshambaugh** [comment]
Bump on this now that main is back open

**Comment by scottshambaugh** [comment]
Bump on merging this now that 3.8.0 is out

**Review by oscargus — APPROVED** [review]

**Review by timhoffm — APPROVED** [review]

**Comment by timhoffm** [comment]
@scottshambaugh Thanks!

**Comment by ksunden** [comment]
Unclear to me why this didn't happen for this PR, but references to `[xy]axis_inverted` and `invert_[xy]axis` in the docstrings from this PR are causing doc build failures on main now:

https://app.circleci.com/pipelines/github/matplotlib/matplotlib/26910/workflows/8053f52d-d0f9-45f0-8b9e-e759ea3c4215/jobs/78885

```
/home/circleci/project/lib/matplotlib/axes/_base.py:docstring of matplotlib.axes._base._AxesBase.get_xbound:20: WARNING: py:obj reference target not found: invert_xaxis
/home/circleci/project/lib/matplotlib/axes/_base.py:docstring of matplotlib.axes._base._AxesBase.get_xbound:20: WARNING: py:obj reference target not found: xaxis_inverted
/home/circleci/project/lib/matplotlib/axes/_base.py:docstring of matplotlib.axes._base._AxesBase.get_ybound:20: WARNING: py:obj reference target not found: invert_yaxis
/home/circleci/project/lib/matplotlib/axes/_base.py:docstring of matplotlib.axes._base._AxesBase.get_ybound:20: WARNING: py:obj reference target not found: yaxis_inverted
/home/circleci/project/lib/mpl_toolkits/mplot3d/axes3d.py:docstring of mpl_toolkits.mplot3d.axes3d.Axes3D.set_xbound:31: WARNING: py:obj reference target not found: invert_xaxis
/home/circleci/project/lib/mpl_toolkits/mplot3d/axes3d.py:docstring of mpl_toolkits.mplot3d.axes3d.Axes3D.set_xbound:31: WARNING: py:obj reference target not found: xaxis_inverted
/home/circleci/project/lib/mpl_toolkits/mplot3d/axes3d.py:docstring of mpl_toolkits.mplot3d.axes3d.Axes3D.set_xlim:53: WARNING: py:obj reference target not found: invert_xaxis
/home/circleci/project/lib/mpl_toolkits/mplot3d/axes3d.py:docstring of mpl_toolkits.mplot3d.axes3d.Axes3D.set_xlim:53: WARNING: py:obj reference target not found: xaxis_inverted
/home/circleci/project/lib/mpl_toolkits/mplot3d/axes3d.py:docstring of mpl_toolkits.mplot3d.axes3d.Axes3D.set_ybound:31: WARNING: py:obj reference target not found: invert_yaxis
/home/circleci/project/lib/mpl_toolkits/mplot3d/axes3d.py:docstring of mpl_toolkits.mplot3d.axes3d.Axes3D.set_ybound:31: WARNING: py:obj reference target not found: yaxis_inverted
/home/circleci/project/lib/mpl_toolkits/mplot3d/axes3d.py:docstring of mpl_toolkits.mplot3d.axes3d.Axes3D.set_ylim:53: WARNING: py:obj reference target not found: invert_yaxis
/home/circleci/project/lib/mpl_toolkits/mplot3d/axes3d.py:docstring of mpl_toolkits.mplot3d.axes3d.Axes3D.set_ylim:53: WARNING: py:obj reference target not found: yaxis_inverted
```


It looks like the warning is that it can't find them _at all_, rather than that it e.g. sees both `Axes.invert_yaxis` and `Axes3D.invert_yaxis`, which was my first thought.

Sphinx version is different (7.2.4 here, 7.2.6 on main), though only as patch releases, and nothing obviously related in release notes.

`mpl-sphinx-theme` version is different, though I can't think of a way that should affect this linking behavior.
